src/glmtools.jl

```diff
@@ -419,6 +431,10 @@ function devresid(::Binomial, y, μ::Real)
     end
 end
 devresid(::Gamma, y, μ::Real) = -2 * (log(y / μ) - (y - μ) / μ)
+function devresid(::Geometric, y, μ::Real)
+    v = 2 * (xlogy(y, y / μ) + xlogy(y + 1, (μ + 1) / (y + 1)))
```

Comment:
The notes I have for squared deviance residuals are:

```julia
    v = 2 * (xlogy(y, y / μ) - xlogy(y + 1, (y + 1) / (μ + 1)))
```

but this is equivalent to the operation here. I have no idea which version will tend to perform better numerically in practice.

Reply:
Looks like the following one is little faster
`2 * (xlogy(y, y / μ) - xlogy(y + 1, (y + 1) / (μ + 1)))`
Updated the code accordingly.